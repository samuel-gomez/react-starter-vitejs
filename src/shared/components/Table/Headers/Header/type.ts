import { ReactNode } from 'react';

export type THeader = {
  label: string;
  key: string;
  field?: string;
  infobulle?: ReactNode;
};
