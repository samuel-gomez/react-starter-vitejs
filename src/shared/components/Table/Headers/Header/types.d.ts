import { type ReactNode } from 'react';

export type THeader = {
  key: string;
  label?: string;
  field?: string;
  infobulle?: ReactNode;
  isHeader?: boolean;
};
