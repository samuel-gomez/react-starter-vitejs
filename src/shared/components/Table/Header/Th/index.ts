export { default } from './Th.container';
export type { TThContainer } from './Th.container';
export type { TsetSort, Torder } from './ThSortable';
