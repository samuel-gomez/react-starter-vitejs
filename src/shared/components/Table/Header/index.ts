export { default } from './Header';
export type { THeader } from './Header';
