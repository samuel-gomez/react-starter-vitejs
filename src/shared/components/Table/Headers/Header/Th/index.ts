export { default } from './Th';
export type { TTh } from './Th';
export type { TsetSort, Torder } from './ThSortable';
