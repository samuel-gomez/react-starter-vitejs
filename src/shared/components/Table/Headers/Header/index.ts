export { default, type THeaderProps } from './Header';
export { type THeader } from './types.d';
