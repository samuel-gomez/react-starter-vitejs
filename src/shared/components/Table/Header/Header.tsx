import type { ReactNode } from 'react';
import { Table as TableTk, HelpInfo } from '@axa-fr/react-toolkit-all';
import Th, { TTh } from './Th';

type Theaders = {
  label: string;
  key: string;
  field?: string;
  infobulle?: ReactNode;
};

export type THeader = TTh & {
  headers?: Theaders[];
  children?: ReactNode;
  ariaLabel?: string;
  ariaLabelLine?: string;
};

const Header = ({ headers = [], onSort, sorting, children, ariaLabel = 'table-header', ariaLabelLine = 'table-header-line' }: THeader) => (
  <TableTk.Header className="af-table__thead" aria-label={ariaLabel}>
    <TableTk.Tr aria-label={ariaLabelLine}>
      <>
        {!!headers.length &&
          headers.map(({ field, label, key, infobulle }) => (
            <Th key={key} scope="col" sorting={sorting} field={field} onSort={onSort}>
              <HelpInfo content={infobulle}>
                <span className="af-table__th-label">{label}</span>
              </HelpInfo>
            </Th>
          ))}
        {children}
      </>
    </TableTk.Tr>
  </TableTk.Header>
);

export default Header;
