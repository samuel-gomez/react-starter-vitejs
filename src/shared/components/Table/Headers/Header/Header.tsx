import { HelpInfo } from '@axa-fr/react-toolkit-all';
import Th, { type TTh } from './Th';
import { type THeader } from './types.d';

export type THeaderProps = TTh &
  Omit<THeader, 'key'> & {
    scope?: 'col' | 'row';
    isBlank?: boolean;
  };

const Header = ({ field, infobulle, label, scope = 'col', onSort, sorting, isBlank = false, ...rest }: THeaderProps) => (
  <Th {...(isBlank ? { classModifier: 'blank' } : { scope, sorting, field, onSort, ...rest })}>
    {!isBlank && (
      <HelpInfo content={infobulle}>
        <span className="af-table__th-label">{label}</span>
      </HelpInfo>
    )}
  </Th>
);

export default Header;
