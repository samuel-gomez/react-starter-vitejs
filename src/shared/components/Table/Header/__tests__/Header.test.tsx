import { describe, it, expect, vi } from 'vitest';
import { renderWithContainer } from 'shared/testsUtils';
import Header from '../Header';

const defaultProps = {
  onSort: vi.fn(),
};

const container = document.createElement('table');

const headersMock = [
  {
    label: 'label',
    key: 'uid',
    field: 'name',
  },
];
const sortingMock = {
  field: 'name',
  order: 1,
};

describe('Header', () => {
  it.each`
    headers        | sorting        | children
    ${undefined}   | ${undefined}   | ${undefined}
    ${[]}          | ${undefined}   | ${undefined}
    ${headersMock} | ${undefined}   | ${undefined}
    ${headersMock} | ${sortingMock} | ${undefined}
    ${undefined}   | ${sortingMock} | ${undefined}
    ${headersMock} | ${sortingMock} | ${(<th>child header</th>)}
  `('Should render Header when headers: $headers, sorting: $sorting', ({ headers, sorting, children }) => {
    const { baseElement } = renderWithContainer(
      <Header {...defaultProps} headers={headers} sorting={sorting}>
        {children}
      </Header>,
      container,
    );
    expect(baseElement).toMatchSnapshot();
  });
});
