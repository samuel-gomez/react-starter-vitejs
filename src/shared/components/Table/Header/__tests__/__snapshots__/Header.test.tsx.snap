// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Header > Should render Header when headers: [ Array(1) ], sorting: { field: 'name', order: 1 } 1`] = `
<table>
  <thead
    aria-label="table-header"
    class="af-table__thead"
  >
    <tr
      aria-label="table-header-line"
      class="af-table__tr"
    >
      <th
        class="af-table__th af-table__th--sortable af-table__th--active"
        role="button"
      >
        <span
          class="af-table__th-label"
        >
          label
        </span>
        <span
          class="af-btn__icon af-btn__icon--table-sorting glyphicon glyphicon-arrow-xs-up"
        />
      </th>
    </tr>
  </thead>
</table>
`;

exports[`Header > Should render Header when headers: [ Array(1) ], sorting: { field: 'name', order: 1 } 2`] = `
<table>
  <thead
    aria-label="table-header"
    class="af-table__thead"
  >
    <tr
      aria-label="table-header-line"
      class="af-table__tr"
    >
      <th
        class="af-table__th af-table__th--sortable af-table__th--active"
        role="button"
      >
        <span
          class="af-table__th-label"
        >
          label
        </span>
        <span
          class="af-btn__icon af-btn__icon--table-sorting glyphicon glyphicon-arrow-xs-up"
        />
      </th>
      <th>
        child header
      </th>
    </tr>
  </thead>
</table>
`;

exports[`Header > Should render Header when headers: [ Array(1) ], sorting: undefined 1`] = `
<table>
  <thead
    aria-label="table-header"
    class="af-table__thead"
  >
    <tr
      aria-label="table-header-line"
      class="af-table__tr"
    >
      <th
        class="af-table__th af-table__th--sortable af-table__th--active"
        role="button"
      >
        <span
          class="af-table__th-label"
        >
          label
        </span>
        <span
          class="af-btn__icon af-btn__icon--table-sorting glyphicon glyphicon-sorting"
        />
      </th>
    </tr>
  </thead>
</table>
`;

exports[`Header > Should render Header when headers: [], sorting: undefined 1`] = `
<table>
  <thead
    aria-label="table-header"
    class="af-table__thead"
  >
    <tr
      aria-label="table-header-line"
      class="af-table__tr"
    />
  </thead>
</table>
`;

exports[`Header > Should render Header when headers: undefined, sorting: { field: 'name', order: 1 } 1`] = `
<table>
  <thead
    aria-label="table-header"
    class="af-table__thead"
  >
    <tr
      aria-label="table-header-line"
      class="af-table__tr"
    />
  </thead>
</table>
`;

exports[`Header > Should render Header when headers: undefined, sorting: undefined 1`] = `
<table>
  <thead
    aria-label="table-header"
    class="af-table__thead"
  >
    <tr
      aria-label="table-header-line"
      class="af-table__tr"
    />
  </thead>
</table>
`;
